src/ripple/app/tx/impl/CancelOffer.cpp

```diff
@@ -39,10 +39,26 @@ CancelOffer::preflight(PreflightContext const& ctx)
         return temINVALID_FLAG;
     }
 
-    if (!ctx.tx[sfOfferSequence])
+    if (ctx.rules.enabled(fixXahauV1))
     {
-        JLOG(ctx.j.trace()) << "CancelOffer::preflight: missing sequence";
-        return temBAD_SEQUENCE;
+        if ((!ctx.tx.isFieldPresent(sfOfferSequence) &&
+             !ctx.tx.isFieldPresent(sfOfferID)) ||
+            (ctx.tx.isFieldPresent(sfOfferSequence) &&
+             ctx.tx.isFieldPresent(sfOfferID)))
+        {
+            JLOG(ctx.j.trace())
+                << "CancelOffer::preflight: invalid sequence or offer id";
+            return temBAD_SEQUENCE;
+        }
+    }
+    else
+    {
+        if (!ctx.tx.isFieldPresent(sfOfferSequence) ||
+            ctx.tx[sfOfferSequence] == 0)
```

Comment:
condition should match exactly what it was `if (!ctx.tx[sfOfferSequence])` otherwise if the behaviour is even slightly different this can fork

Reply:
This was written like this because `sfOfferSequence` is now optional. So I think `ctx.tx[sfOfferSequence]` fails hard. I can revert and show the output.